MAN: Document different defaults for IPA and AD providers

justin-stephenson opened the pull request:
Update man pages for any AD provider config options that differ from
ldap/krb5 provider back-end defaults.

Resolves:
https://fedorahosted.org/sssd/ticket/3214

I would appreciate any suggestions on improving the wording, I was hoping to be informative but concise.

sumit-bose commented:
ok to test

jhrozek commented:
Thank you for the patch. I checked it by inspecting the values from ad_opts.c and krb5_opts.c and ldap_opts.c with vimdiff. I only have two questions:

1) Should we do the same for the IPA provider as a separate patch?

2) The sssd-ad man page still lists `krb5_use_enterprise_principal` and `krb5_confd_path`. I don't think we need to include them as long as the options are listed with their ad-specific option in the new section (that's for `krb5_confd_path`) or if the option just uses the default (`krb5_confd_path`). I think this could be an additional patch atop this one.

justin-stephenson commented:
Thanks for your comments Jakub, please see below:

1) Yes, I will go ahead and submit a different patch in this same PR if it's okay.

2) I agree about removing `krb5_use_enterprise_principal` because it will be included in the new section and the option explanation can be found in the `sssd-krb5(5)` man page, however the `krb5_confd_path` option still needs to have the option details explained somewhere. Do you think it should it be added in the new section or left as-is ?

jhrozek commented:
On Wed, Nov 02, 2016 at 06:18:57AM -0700, Justin Stephenson wrote:
> Thanks for your comments Jakub, please see below:
> 
> 1) Yes, I will go ahead and submit a different patch in this same PR if it's okay.
> 
> 2) I agree about removing `krb5_use_enterprise_principal` because it will be included in the new section and the option explanation can be found in the `sssd-krb5(5)` man page, however the `krb5_confd_path` option still needs to have the option details explained somewhere. Do you think it should it be added in the new section or left as-is ?

Ah, you are right about krb5_confd_path, this option should be left
as-is I think. Thanks for catching that.

justin-stephenson commented:
- Revised the first patch to remove `krb5_use_enterprise_principal` from the sssd-ad man page.
- Added IPA provider modified defaults patch.
- Removed `krb5_validate`, `krb5_use_fast`, and `krb5_canonicalize` from the sssd-ipa man page as they are now mentioned in the modified defaults(with option explanation still in sssd-krb5).

Note: The IPA provider sets **ldap_default_authtok_type = NULL** and **ldap_access_order` = NULL** but I did not include these in the list.

lslebodn commented:
Could you add newly created xml files into src/man/po/po4a.cfg ?

LS

lslebodn commented:
I can see that tokengroups are documented in this PR. So I think we can close upstream ticket https://fedorahosted.org/sssd/ticket/3233 as a duplicate of https://fedorahosted.org/sssd/ticket/3214

jhrozek commented:
Originally I was thinking about adding a separate sentence like we have for describing case-sensitiveness:
```
Users, groups and other entities served by SSSD are always treated as case-insensitive in the AD provider for compatibility with Active Directory's LDAP implementation.
```
More or less to have more visibility. I'm afraid not many administrators will even realize what tokengroups is otherwise.

jhrozek commented:
Either way, I think these patches can be re-reviewed and pushed separately.

jhrozek commented:
ACK. I won't push the patches until tomorrow though in case @lslebodn or someone else wanted to take a look as well. I'll also wait for CI results.

jhrozek commented:
master:
6e27e8572f671de575d9ac2a34a677d9efc24fbc
8caf7ba5005b3be5447311713ad2b58169f9d32f
